manifests/agent.pp

```diff
@@ -433,21 +438,26 @@
   }
 
   # Controlling the 'zabbix-agent' service
-  if str2bool(getvar('::systemd')) {
-    $service_provider = 'systemd'
-    $service_path = undef
-  } elsif $facts['os']['name'] == 'AIX' {
-    $service_provider = 'init'
-    $service_path = '/etc/rc.d/init.d'
+  if $service_provider == undef {
+    if str2bool(getvar('::systemd')) {
+      $real_service_provider = 'systemd'
```

Comment:
This is a bit odd. Essentially this gets `$facts['systemd']` but that is defined as `service_provider == 'systemd'` (https://github.com/camptocamp/puppet-systemd/blob/03d94faacfe51ce95e8ab048f79cfd2617089b36/lib/facter/systemd.rb#L39).

So really, this whole systemd specific block can be removed and handled by the `else` case.

Given that, is the whole change still needed?

Reply:
@bastelfreak and i have been discussing this on IRC, and it really is an odd case: this code shouldn't exist, and this patch shouldn't be necessary

Reply:
@automaticserver could you instead write a patch that removes this code and doesn't introduce a `service_provider` parameter? I don't know about the AIX code, but at least systemd should be removed here. It's probably cleaner as a separate PR.

Reply:
> could you instead write a patch that removes this code and doesn't introduce a service_provider parameter? I don't know about the AIX code, but at least systemd should be removed here. It's probably cleaner as a separate PR.

No.

If the AIX code remains then the module would still explicitly assign `undef` to the `provider` parameter of the service resourece, which would still supersede the default value we're assigning in the global scope. Meaning the problem we've been trying to resolve for three weeks now (and which could've easily been solved by a one-line change) would still persist.

Reply:
I quickly hacked up https://github.com/voxpupuli/puppet-zabbix/pull/711 which I think will solve it. Would be good to get some verification on that though.